### PR TITLE
DAOS-11258 test: Recording launch.py output in logs (#10082)

### PR DESCRIPTION
**Pull request by phender**
Adding logging debug messages for launch.py outut in the Jenkins test
results and including all test post-processing in a separate launch.log
stored with the test results.

Test-tag: pr daily_regression

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

Also includes the following cherry-picked changes:

DAOS-10335 test: move check_pool_free_space() to Testpool class (#8869)

Signed-off-by: Wang Shilong <shilong.wang@intel.com>

DAOS-11762 test: Match log threshold with log rollover size (#10479)

Increasing the test logs threshold size check to match the log rollover
size: 1GB -> 2GB.  Increasing the timeout for the cart_logtest command
and removing any empty log files prior to running the cart_logtest
command. Also resolving DAOS-11812, where non-contiguous node sets where
not processed correctly when updating test yaml files.

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

DAOS-11911 test: Fix launch.py error reporting (#10574)

If launch.py experienced an error before it could establish a file
handler for logging debug messages to a file, it could fail without
providing the reason for the failure.  This could happen when running
manually if the python avocado libraries where not installed or
misconfigured. The console logger will now report all messages until the
file handler is configured, at which point the console logger will be
updated to only report info messages.

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

DAOS-12011 test: setup fuse3.conf (#10774)

Setup both fuse.conf and fuse3.conf to account for OS differences.
Move setup to launch.py

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

DAOS-12049 test: Fix speed check for virtio network interface (#10806)

Now that the interface speed check performed by launch.py is performed
via a clushed cat command the error handling for a virtio network
interface needs to be updated to handle the error coming from the cat
command.  Also fixed the error handling for detecting the core file
pattern.

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficent testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by phender** [comment]
Merging the changes from https://github.com/daos-stack/daos/pull/10621 required manual updates to the following files:

- src/tests/ftest/soak/faults.yaml
- src/tests/ftest/soak/harassers.yaml
- src/tests/ftest/soak/stress_2h.yaml
- src/tests/ftest/soak/stress_48h.yaml
- src/tests/ftest/util/apricot/apricot/test.py
- src/tests/ftest/util/server_utils.py
- src/tests/ftest/util/slurm_utils.py

**Comment by phender** [comment]
There were 3 errors in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10838/3/ that can be accounted for by these tickets:

- https://daosio.atlassian.net/browse/DAOS-12113 (looks like a network issue with wolf-128)
- https://daosio.atlassian.net/browse/DAOS-11925 (known issue)

**Review by mjean308 — APPROVED** [review]
soak changes lgtm

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/launch.py [hunk 1]
```diff
@@ -52,2294 +72,2976 @@
...
+        """
+        logger.info("Setting up fuse config")
+        fuse_configs = ("/etc/fuse.conf", "/etc/fuse3.conf")
+        command = ";".join([
+            "if [ -e {0} ]",
+            "then ls -l {0}",
+            "(grep -q '^user_allow_other$' {0} || echo user_allow_other | sudo tee -a {0})",
+            "cat {0}",
+            "fi"
+        ])
```
[the remark this answers is not in the document]
Comment (phender) [reply src/tests/ftest/launch.py:1755]: Diving into this deeper I didn't realize the initial change to src/tests/ftest/scripts/setup_nodes.sh was part of that feature, so yes this should not be included.

**Comment by phender** [comment]
Seeing as https://github.com/daos-stack/daos/commit/30b6bd29446b916308883a0136e311a4d8ca8048 has landed and I need to remove the https://github.com/daos-stack/daos/commit/3c16ca3d3ceba3440dac161cc3e816276d6f49bd cherry-pick, I will redo this PR.

**Pull request closed without merging**